internal/s3storage/s3storage.go

```diff
@@ -68,3 +163,65 @@ func (s *Service) CreateArtifact(artifactSpec artifact.Spec, md5 string) (string
 
 	return uploadURL, nil
 }
+
+func (s *Service) Close() error {
```

Comment:
The method will panic if called multiple times as we cannot `close` a channel more than once.

It adds event more complexity to handle this, so maybe we should just add a note in the docs for now?
```suggestion
// Close closes the S3 storage service. Multiple calls to this method will result in a panic.
func (s *Service) Close() error {
```

Reply:
Would rather it didn't blow up, but yeah, maybe the comment is good enough in this case.

Reply:
True. Adding a `closed` state that is atomically updated is the usual way, but to keep it simple enough in this controlled API I think we can leave at as is.

Reply:
Ok 👍